### PR TITLE
Introduce new op types `GPI`, `GPI2` and `AAMS`

### LINKED ISSUE #1259: Add native IonQ OpTypes
https://github.com/amazon-braket/amazon-braket-examples/blob/main/examples/braket_features/IonQ_Native_Gates.ipynb

### PR DESCRIPTION
**Pull request by cqc-alec**
# Description

Add the IonQ native ops defined [here](https://github.com/amazon-braket/amazon-braket-examples/blob/main/examples/braket_features/IonQ_Native_Gates.ipynb).

# Related issues

Closes #1259 .

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Review by yao-cqc — COMMENTED** [review]
Thanks! Only a small problem with the docs

File: pytket/binders/circuit/main.cpp [hunk 1]
```diff
@@ -186,6 +186,28 @@ PYBIND11_MODULE(circuit, m) {
...
+          "\\begin{array}{cccc} \\cos\\frac{\\pi\\theta}{2} & 0 & 0 & "
+          "-ie^{-i\\pi{\\phi_0+\\phi_1}}\\sin\\frac{\\pi\\theta}{2} \\\\ "
+          "0 & "
+          "\\cos\\frac{\\pi\\theta}{2} & "
+          "-ie^{i\\pi{\\phi_0-\\phi_1}}\\sin\\frac{\\pi\\theta}{2} & 0 \\\\ 0 "
+          "& "
+          "-ie^{i\\pi{\\phi_0-\\phi_1}}\\sin\\frac{\\pi\\theta}{2} & "
+          "\\cos\\frac{\\pi\\theta}{2} & 0 \\\\ "
+          "-ie^{i\\pi{\\phi_0+\\phi_1}}\\sin\\frac{\\pi\\theta}{2} & 0 & 0 & "
+          "\\cos\\frac{\\pi\\theta}{2} \\end{array} \\right]`")
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply pytket/binders/circuit/main.cpp]: Good spot, thanks.

**Review by yao-cqc — APPROVED** [review]

**Comment by cqc-alec** [comment]
PR has been merged, but PR status not updated for some reason.

**Comment by cqc-alec** [comment]
Closing this manually as github problems prevented it from being closed when merged.

**Pull request closed without merging**